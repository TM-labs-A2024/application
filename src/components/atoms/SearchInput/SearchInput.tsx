import { SearchIcon } from '@chakra-ui/icons'
import { InputGroup, Input, InputLeftElement } from '@chakra-ui/react'
import React from 'react'

export default function SearchInputComponent({
  placeholder,
  className,
  disabled,
  onChange = () => null,
  onClick,
  inputRef
}: {
  placeholder: string
  className?: string
  disabled?: boolean
  onChange?: (value: string) => void
  onClick?: () => void
  inputRef?: React.RefObject<HTMLInputElement>
}) {
  return (
    <div className={className}>
      <InputGroup>
        <InputLeftElement pointerEvents="none">
          <SearchIcon color="gray.500" />
        </InputLeftElement>
        <Input
          className="min-h-10"
          onFocus={onClick}
          ref={inputRef}
          placeholder={placeholder}
          disabled={disabled}
          onChange={(e) => onChange(e.target.value)}
        />
      </InputGroup>
    </div>
  )
}
