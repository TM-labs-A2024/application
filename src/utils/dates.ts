export const formatDate = (date: string) => {
  const chunks = date.split('-')

  const newDate = chunks[1] + '-' + chunks[2] + '-' + chunks[0]

  return newDate
}
