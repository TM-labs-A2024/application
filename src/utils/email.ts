import emailjs from '@emailjs/browser'

export function sendEmail(data: {
  from_name: string
  to_name: string
  code: string | number
  to_email: string
}) {
  const templateParams = {
    ...data
  }

  const serviceKey = process.env.NEXT_PUBLIC_EMAIL_SERVICE_KEY ?? ''
  const templateKey = process.env.NEXT_PUBLIC_EMAIL_TEMPLATE_KEY ?? ''
  const apiKey = process.env.NEXT_PUBLIC_EMAIL_API_KEY ?? ''

  emailjs.send(serviceKey, templateKey, templateParams, apiKey).catch(() => {
    alert('Algo sali√≥ mal, por favor reintenta el registro.')
  })
}
