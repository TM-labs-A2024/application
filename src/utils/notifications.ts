import { GENERIC_ERROR } from '@constants/index'
import { isMobile } from '@utils/index'
import { Store } from 'react-notifications-component'

export const setupErrorNotification = () => {
  Store.addNotification(
    GENERIC_ERROR('Algo salió mal, por favor inténtelo de nuevo.', isMobile(window))
  )
}
