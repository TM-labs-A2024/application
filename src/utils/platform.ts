import { Capacitor } from '@capacitor/core'

export const isWeb = () => Capacitor.getPlatform() === 'web'

export const isIOS = () => Capacitor.getPlatform() === 'ios'

export const isAndroid = () => Capacitor.getPlatform() === 'android'

export const isMobile = (window: { screen: { availWidth: number } }) =>
  Capacitor.getPlatform() === 'ios' ||
  Capacitor.getPlatform() === 'android' ||
  window?.screen?.availWidth < 425
