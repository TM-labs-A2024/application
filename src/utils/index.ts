import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
