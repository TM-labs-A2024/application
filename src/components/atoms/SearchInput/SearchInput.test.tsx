import { render, screen, fireEvent } from '@test/utils/index'

import '@testing-library/jest-dom'
import SearchInput from './SearchInput'

const props = {
  placeholder: 'Placeholder text',
  label: 'Label',
  className: '',
  disabled: false,
  id: ''
}

const disabledProps = {
  ...props,
  disabled: true
}

describe('Atoms > SearchInput test', () => {
  test("The component shows the placeholder text when it's empty", () => {
    render(<SearchInput {...props} />)

    expect(screen.getByPlaceholderText(props.placeholder)).toBeInTheDocument()
  })

  test('The component shows the written text when receiving input', () => {
    render(<SearchInput {...props} />)

    const input = screen.getByPlaceholderText(props.placeholder) as HTMLInputElement
    fireEvent.change(input, { target: { value: 'Test text' } })

    expect(input.value).toBe('Test text')
  })

  test('The component is disabled when the disabled prop is true', async () => {
    render(<SearchInput {...disabledProps} />)

    expect(screen.getByPlaceholderText(disabledProps.placeholder)).toBeDisabled()
  })

  test('Matches the snapshot', async () => {
    render(<SearchInput {...props} />)

    expect(await screen.findByTestId('search-input-component')).toMatchSnapshot()
  })
})
