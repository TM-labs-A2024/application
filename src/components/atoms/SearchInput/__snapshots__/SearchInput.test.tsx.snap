// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms > SearchInput test Matches the snapshot 1`] = `
<div
  class=""
  data-testid="search-input-component"
>
  <div
    class="chakra-input__group css-1y0e7gb"
    data-group="true"
  >
    <div
      class="chakra-input__left-element css-gv1iwz"
    >
      <svg
        class="chakra-icon css-13izzit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
          fill="currentColor"
        />
      </svg>
    </div>
    <input
      class="chakra-input min-h-10 css-qluo93"
      placeholder="Placeholder text"
    />
  </div>
</div>
`
